File: src/commands/configure.cpp
#include "configure.h"
#include "../core/config.h"
#include "../core/project.h"
#include <iostream>
#include <stdexcept>

namespace sugar {

ConfigureCommand::ConfigureCommand() : Command("configure") {
}

int ConfigureCommand::execute(const std::string& config_path) {
    try {
        std::string cfg_path = config_path.empty() ? "sugar.toml" : config_path;
        std::cout << "Configuring from: " << cfg_path << std::endl;

        Config config = Config::load(cfg_path);
        config.validate();

        Project project(config);

        std::cout << "Configuration validation successful!" << std::endl;
        std::cout << "  Project: " << config.project_name << std::endl;
        std::cout << "  Type: " << config.project_type << std::endl;
        std::cout << "  Compiler: " << config.compiler << std::endl;
        std::cout << "  Platform: " << config.platform << std::endl;
        std::cout << "  Source paths: ";
        for (size_t i = 0; i < config.source_paths.size(); ++i) {
            if (i > 0) std::cout << ", ";
            std::cout << config.source_paths[i];
        }
        std::cout << std::endl;
        std::cout << "  Build path: " << config.build_path << std::endl;
        std::cout << "  Output path: " << config.output_path << std::endl;
        if (!config.link_dependencies.empty()) {
            std::cout << "  Dependencies: ";
            for (size_t i = 0; i < config.link_dependencies.size(); ++i) {
                if (i > 0) std::cout << ", ";
                std::cout << config.link_dependencies[i];
            }
            std::cout << std::endl;
        }

        return 0;
    } catch (const std::exception& e) {
        std::cerr << "Error: " << e.what() << std::endl;
        return 1;
    }
}

std::string ConfigureCommand::getHelp() const {
    return R"(
configure - Validate sugar.toml configuration

Usage: sugar-builder configure [--config <path>]

Options:
  --config <path>    Path to sugar.toml (defaults to ./sugar.toml)

Description:
  Loads and validates the sugar.toml configuration file without building.
  Checks for required fields and validates project settings.
)";
}

} // namespace sugar

File: python version/commands/configure.py
"""Configure command for SugarBuilder."""

from pathlib import Path
from typing import Optional
from .base import Command
from src.core import Config, Project


class ConfigureCommand(Command):
    """
    Configure command validates the sugar.toml configuration.
    
    Loads and validates project configuration without building.
    """
    
    def __init__(self):
        """Initialize configure command."""
        super().__init__("configure")
    
    def execute(self, config_path: Optional[str] = None) -> int:
        """
        Load and validate configuration.
        
        Args:
            config_path: Optional path to sugar.toml (defaults to ./sugar.toml).
            
        Returns:
            0 on success, 1 on failure.
        """
        try:
            # Default to ./sugar.toml if not specified
            if config_path is None:
                config_path = "sugar.toml"
            
            print(f"Configuring from: {config_path}")
            
            # Load configuration
            config = Config.load(config_path)
            
            # Validate configuration
            config.validate()
            
            # Create project to verify paths
            project = Project(config)
            
            print("Configuration validation successful!")
            print(f"  Project: {config.project_name}")
            print(f"  Type: {config.project_type}")
            print(f"  Compiler: {config.compiler}")
            print(f"  Platform: {config.platform}")
            print(f"  Source paths: {', '.join(config.source_paths)}")
            print(f"  Build path: {config.build_path}")
            print(f"  Output path: {config.output_path}")
            if config.link_dependencies:
                print(f"  Dependencies: {', '.join(config.link_dependencies)}")
            
            return 0
        
        except FileNotFoundError as e:
            print(f"Error: {e}")
            return 1
        except ValueError as e:
            print(f"Configuration Error: {e}")
            return 1
        except Exception as e:
            print(f"Unexpected error: {e}")
            return 1
    
    def get_help(self) -> str:
        """Get help text for configure command."""
        return """
configure - Validate sugar.toml configuration

Usage: sugar-builder configure [--config <path>]

Options:
  --config <path>    Path to sugar.toml (defaults to ./sugar.toml)

Description:
  Loads and validates the sugar.toml configuration file without building.
  Checks for required fields and validates project settings.
"""
